src/rust/ide/lib/view/src/graph_editor/component/node/port.rs

```diff
@@ -140,6 +143,21 @@ impl Expression {
         let output_span_tree = default();
         Self {code,input_span_tree,output_span_tree}
     }
+
+    fn get_id_for_crumbs(&self, crumbs:&[span_tree::Crumb]) -> Option<ast::Id> {
+        // We don't know whether this is from the input or output span tree, so we need to
+        // check both.
+        let span_trees = [&self.input_span_tree, &self.output_span_tree];
```

Comment:
IMO the sae effecet would be when naming the var `span_trees_to_check`. You can leave comment if you want, but I'm trying to remove form code all non-essential comments (comments which cannot be explained by good var names), like comments about some system depended choices (e.g. "On Windows we need this because this and that").